src/grappler/ngraph_optimizer.cc

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_optimizer.h"
+
+#include "tensorflow/core/framework/attr_value.pb.h"
+#include "tensorflow/core/framework/node_def.pb.h"
+#include "tensorflow/core/grappler/clusters/cluster.h"
+#include "tensorflow/core/grappler/grappler_item.h"
+#include "tensorflow/core/grappler/op_types.h"
+#include "tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.h"
+#include "tensorflow/core/grappler/utils.h"
+#include "tensorflow/core/platform/protobuf.h"
+
+#include <iomanip>
+#if defined NGRAPH_DISTRIBUTED
+#include "ngraph/distributed.hpp"
+#endif
+
+#include <iostream>
+
+using namespace std;
+
+namespace tensorflow {
+namespace ngraph_bridge {
+
+Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
+                                 const tensorflow::grappler::GrapplerItem& item,
+                                 GraphDef* output) {
+  NGRAPH_VLOG(1) << "Here at NgraphOptimizer ";
+
+  // Convert the GraphDef to Graph
```

Comment:
Move this block after line 84 i.e., after checking whether to continue with nGraph.

Reply:
Moved the check to the very top.

Reply:
Ok. I will need to move the check for NGraph enable back to where it was since in this case we need to get the graph from the graphdef to be able to dump it.